### ISSUE TITLE
SE Writer priority issue when using static content

### ISSUE THREAD

**Issue by tomas-langer** [oracle/helidon#1585]
2.0 `master`

When I configure routing as follows:
```
                .register(health) // Health at "/health"
                .register(metrics) // Metrics at "/metrics"
                .register(JsonBindingSupport.create())
                .register("/employees", employeeService)
                .register("/public", StaticContentSupport.builder("public")
                        .welcomeFileName("index.html"))
                .build();
```

the static content support does not work.
This is caused by discovery of writers - the JSON-B writer accepts any type, including `Path`.
That means the default `Path` writer is not triggered, and instead of serving the resource from file system (even though this is classpath, we use file `Path` if served from path - running from IDE that uses `target/classes`), it tried to create a JSON from `Path`.

**Comment by romain-grecourt** [comment]
This is working as designed, the employee-app example is a corner case of writers ordering with a Writer (`JsonbBodyWriter`) that accepts everything and trumps the built-in writer for `Path`.

The following changes the employee-app make it work:
```diff
diff --git a/examples/employee-app/src/main/java/io/helidon/service/employee/Main.java b/examples/employee-app/src/main/java/io/helidon/service/employee/Main.java
index 94764db63..ad539ff41 100644
--- a/examples/employee-app/src/main/java/io/helidon/service/employee/Main.java
+++ b/examples/employee-app/src/main/java/io/helidon/service/employee/Main.java
@@ -20,10 +20,15 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.util.logging.LogManager;
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+
 import io.helidon.config.Config;
 import io.helidon.health.HealthSupport;
 import io.helidon.health.checks.HealthChecks;
-import io.helidon.media.jsonb.server.JsonBindingSupport;
+import io.helidon.media.common.MediaSupport;
+import io.helidon.media.jsonb.common.JsonbBodyReader;
+import io.helidon.media.jsonb.common.JsonbBodyWriter;
 import io.helidon.metrics.MetricsSupport;
 import io.helidon.webserver.Routing;
 import io.helidon.webserver.ServerConfiguration;
@@ -68,7 +73,10 @@ public final class Main {
         // Get webserver config from the "server" section of application.yaml
         ServerConfiguration serverConfig = ServerConfiguration.create(config.get("server"));
-        WebServer server = WebServer.create(serverConfig, createRouting(config));
+        WebServer server = WebServer.builder(createRouting(config))
+                .mediaSupport(createMediaSupport())
+                .config(serverConfig)
+                .build();
         // Try to start the server. If successful, print some info and arrange to
         // print a message at shutdown. If unsuccessful, print the exception.
@@ -88,9 +96,22 @@ public final class Main {
         return server;
     }
+    /**
+     * Create new {@link MediaSupport}.
+     * @return media support configured with JSON binding support
+     */
+    private static MediaSupport createMediaSupport() {
+        Jsonb json = JsonbBuilder.create();
+        return MediaSupport.builder()
+                .registerReader(JsonbBodyReader.create(json))
+                .registerWriter(JsonbBodyWriter.create(json))
+                .registerDefaults()
+                .build();
+    }
+
     /**
      * Creates new {@link Routing}.
-     * @return routing configured with JSON support, a health check, and a service
+     * @return routing configured with a health check, and a service
      * @param config configuration of this server
      */
     private static Routing createRouting(Config config) {
@@ -100,7 +121,7 @@ public final class Main {
         HealthSupport health = HealthSupport.builder().add(HealthChecks.healthChecks())
                 .build(); // Adds a convenient set of checks
-        return Routing.builder().register(JsonBindingSupport.create())
+        return Routing.builder()
                 .register(health) // Health at "/health"
                 .register(metrics) // Metrics at "/metrics"
                 .register("/employees", employeeService)
```

This raises the question of media operator order sensitivity for our built-in handlers such as the static content support metrics and health check. Will file a separate issue to track this.

**Comment by romain-grecourt** [comment]
See https://github.com/oracle/helidon/issues/1588

**Comment by tomas-langer** [comment]
This issue was resolved by #1899 and other media support PRs.
